### PR TITLE
[PF-2977] Azure Service Bus support for stairway

### PR DESCRIPTION
**Pull request by v-hsompalli**
1) Updated common-lib build file to use the latest version of stairway which supports Azure Service Bus(0.0.79)
2) Updated properties file to include Azure Service Bus related properties, **azureQueueEnabled** property should be true to use azure service bus queue as Work Queue. If it set to **false** so that gcp would be the Work Queue.
3) setupWorkQueue()  method from StairwayComponent is updated to return QueueInterface ,  so it will suport both GCP and Azure.

### CODE REVIEW

**Review by giventocode — APPROVED** [review]
LGTM

File: build.gradle [hunk 1]
```diff
@@ -81,6 +81,7 @@ dependencies {
     // Terra libraries
     implementation group: 'org.broadinstitute.dsde.workbench', name: 'sam-client_2.13', version: '0.1-f554115'
     api group: 'bio.terra', name: 'stairway-gcp', version: '0.0.76-SNAPSHOT'
+    implementation group: 'bio.terra', name: 'stairway-azure', version: '0.0.79-SNAPSHOT'
```
Comment (dvoet) [inline build.gradle]: stairwayy-gcp and stairway-azure should use the same version, define it as a constant

File: src/main/java/bio/terra/common/stairway/StairwayComponent.java [hunk 2]
```diff
@@ -54,7 +57,7 @@ public StairwayComponent(
    * clusterNameSuffix is present, then we create the queue. It is an error to try to run in
    * Kubernetes without a work queue.
    */
-  private GcpPubSubQueue setupWorkQueue() {
+  private QueueInterface setupWorkQueue() {
```
Comment (dvoet) [inline src/main/java/bio/terra/common/stairway/StairwayComponent.java]: rename to setupGcpWorkQueue?

File: src/main/java/bio/terra/common/stairway/StairwayComponent.java [hunk 3]
```diff
@@ -91,6 +94,21 @@ private GcpPubSubQueue setupWorkQueue() {
     }
   }
 
+
+  /**
+   * Set up the Stairway Azure work queue.
+   */
+  public QueueInterface setupAzureWorkQueue() {
```
Comment (dvoet) [inline src/main/java/bio/terra/common/stairway/StairwayComponent.java]:
```suggestion
  private QueueInterface setupAzureWorkQueue() {
```

File: src/main/java/bio/terra/common/stairway/StairwayComponent.java [hunk 4]
```diff
@@ -113,7 +131,13 @@ public void initialize(DataSource dataSource, Object context, List<StairwayHook>
    * @param initializeBuilder collection of Stairway initialization parameters
    */
   public void initialize(StairwayOptionsBuilder initializeBuilder) {
-    QueueInterface queue = (kubeProperties.isInKubernetes()) ? setupWorkQueue() : null;
+    QueueInterface queue;
+    //Using Azure WorkQueue if azureQueueEnabled set to true
+    if(stairwayProperties.isAzureQueueEnabled()) {
+      queue = setupAzureWorkQueue();
```
Comment (dvoet) [inline src/main/java/bio/terra/common/stairway/StairwayComponent.java:132]: why does gcp care about kubeProperties.isInKubernetes but azure does not?

Comment (v-hsompalli) [reply src/main/java/bio/terra/common/stairway/StairwayComponent.java:132]: This is existing code and Im not sure what is the reason behind it.

Comment (dvoet) [reply src/main/java/bio/terra/common/stairway/StairwayComponent.java:132]: I am suggesting you find out the reason. It is weird to me that gcp would care about k8s but azure would not.

File: src/main/java/bio/terra/common/stairway/StairwayComponent.java (hunk 3 above)
Comment (dvoet) [reply src/main/java/bio/terra/common/stairway/StairwayComponent.java]: if this is used in tests, make this package private and use the `VisibleForTesting` annotation

**Review by dvoet — APPROVED** [review]

**Review by giventocode — APPROVED** [review]

**Review by giventocode — APPROVED** [review]

**Review by dvoet — APPROVED** [review]

**Review by dvoet — APPROVED** [review]

**Review by giventocode — APPROVED** [review]

**Review by rtitle — APPROVED** [review]
Looks good to me

**Pull request merged**